x-pack/plugins/alerting/public/services/maintenance_windows_api/find.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { HttpSetup } from '@kbn/core/public';
+import { AsApiContract, RewriteRequestCase } from '@kbn/actions-plugin/common';
+
+import { MaintenanceWindowFindResponse } from '../../pages/maintenance_windows/types';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../../common';
+
+const rewriteBodyRes = (results: {
+  data: Array<AsApiContract<MaintenanceWindowFindResponse>>;
+  total: number;
+}): MaintenanceWindowFindResponse[] => {
+  return results.data.map((item) => transform(item));
+};
+
+const transform: RewriteRequestCase<MaintenanceWindowFindResponse> = ({
+  expiration_date: expirationDate,
+  r_rule: rRule,
+  event_start_time: eventStartTime,
+  event_end_time: eventEndTime,
+  created_by: createdBy,
+  updated_by: updatedBy,
+  created_at: createdAt,
+  updated_at: updatedAt,
+  ...rest
+}) => ({
+  ...rest,
+  expirationDate,
+  rRule: { ...rRule },
```

Comment:
curious if we need to make a shallow copy of `rRule`, or we could just change this to `rRule,`

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/154669/commits/b2745dcd188cc86ffc5ecffcfc724d66cc13803f